integration-tests/src/server.js

```diff
@@ -20,26 +21,44 @@ const app = express()
 const server = http.createServer(app)
 server.setTimeout(serverTimeout)
 
+const firestore = new Firestore()
+
 const testTimeout = async timeout => {
   await delay(timeout)
   throw new Error(`Test timeout after ${timeout}ms`)
 }
 
+
+
 const runTest = fn => withPuppeteer(async ({ browser, context, name, page, req, timeout }) => {
   const targetEnvParams = { ...envs[req.query.targetEnv] }
+  const testGroup = req.query.testGroup
```

Comment:
In testing, I got this error when I forgot to specify `testGroup`, and only after the test run completed:

> Error: Value for argument "data" is not a valid Firestore document. Cannot use "undefined" as a Firestore value (found in field "testGroup"). If you want to ignore undefined values, enable `ignoreUndefinedProperties`.

I think we should validate that we have this before running the test or provide a default value if it's missing, like `Date.now()`.